### PR TITLE
Fix JSON argscheme header encoding

### PR DESCRIPTION
**Pull request by abhinav**
Pending confirmation that it was wrong

### CODE REVIEW

**Comment by breerly** [comment]
@blampe cc

**Comment by blampe** [comment]
We agreed arg2 for `as=json` should be `map[string]string`, but no one has updated the protocol to reflect that. Why does this keep coming up?

**Comment by breerly** [comment]
I think regardless of the wireformat, we should only accept `map[string]string` from client-code's POV. @blampe we need to have a quick talk w @Raynos and friends to get on the same page. I had thought that email thread a bit back had us there, but we clearly did something wrong.

**Comment by abhinav** [comment]
@Raynos summarized the current state of things here: https://github.com/uber/tchannel/pull/1391#issuecomment-163455903

**Comment by Raynos** [comment]
@abhinav 

This looks great to me. I'm going to fix node :)

File: tchannel/serializer/json.py [hunk 1]
```diff
@@ -23,14 +23,27 @@
...
     name = JSON
 
-    def __init__(self):
-        pass
+    def serialize_header(self, headers):
+        headers = headers or {}
+
+        for k, v in headers.iteritems():
+            if not (isinstance(k, basestring) and isinstance(v, basestring)):
+                raise ValueError(
```
Comment (breerly) [inline tchannel/serializer/json.py]: nice, thanks

**Comment by breerly** [comment]
Looks good to me. AFAICT you'll need to rebase off of master.

**Pull request merged**